dataset/src/main/scala/frameless/TypedDataset.scala

```diff
@@ -172,18 +172,21 @@ class TypedDataset[T](val dataset: Dataset[T])(implicit val encoder: TypedEncode
   ): GroupedBy2Ops[K1, K2, T] = new GroupedBy2Ops[K1, K2, T](this, c1, c2)
 
   object groupByMany extends ProductArgs {
-    def applyProduct[K <: HList, Out0 <: HList, Out](groupedBy: K)(
+    def applyProduct[K <: HList, Out0 <: HList, Out, Out1](groupedBy: K)(
```

Comment:
You could maybe rename these type parameters to match `GroupedByManyOps[T, TK <: HList, K <: HList, KT]`?